### ISSUE TITLE
DX Enhancement: Improve the ImportTestContent script

### ISSUE THREAD

**Issue by jonathanbossenger** [WordPress/Learn#1864]
Last July, @tellyworth [merged](https://github.com/WordPress/Learn/pull/249) the ImportTestContent script, as a way to get data from Learn WordPress into a local development environment, which was a great improvement over what was available at the time, which was sharing an XML file.

However, it would be ideal if that import script were improved upon, to help with the process of fixing bugs and testing PRs.

I'm opening this issue to discuss some ways to achieve this, and then once there is consensus I suggest we create separate issues for each improvement.

1. Currently the importer script imports 50 pages, tutorials, and lesson plans. However, it does not check if the items being imported already exist in the local database. It would be great if it could either perform this check, to prevent duplicates, or simply wipe the current data and import the new data. The second option would probably be quicker, but it would wipe any data defined locally by the developer.
2. The importer does not import any Topics, and one is required to create a topic and add Tutorials and Lesson plans to that topic manually. It would be ideal if a default topic was created, and all tutorials and lesson plans were assigned to it.
3. The importer does not import any courses. This could be due to one of 2 reasons: a) the course content is not available via the REST API, or b) the course content is available via the REST API, but requires authentication. One possible way to solve this would be to ship a JSON file with one or two small demo courses as part of the codebase and import that. 
4. Depending on the network connection, the importer can time out, resulting in an incomplete import. Currently updating the timeout argument of the wp_remote_get function call to 30 seems to resolve this issue.

**Comment by jonathanbossenger** [comment]
cc @adamwoodnz if you have any thoughts or ideas here.

**Comment by alexstine** [comment]
Totally agreed. I would like a script that mirrors as closely as possible what is actually on Learn production.

**Comment by adamwoodnz** [comment]
👍 All sounds good to me

We should be cautious of investing too much time before the new information architecture for Learning Pathways is decided upon. There is a possibility that some post types might be dropped and the data migrated to Sensei post types (Course, Lesson).

**Comment by jonathanbossenger** [comment]
> I would like a script that mirrors as closely as possible what is actually on Learn production.

@alexstine In terms of the actual data, as Learn WordPress grows, pulling down all the content is going to take more and more time for an import script. So I think a subset of the data is probably a better compromise here.

**Comment by jonathanbossenger** [comment]
@adamwoodnz agreed, which is why for now I'm only suggesting smaller improvements to the script, as this might completely become redundant and have to be recreated once Learning Pathways are implemented.

**Comment by alexstine** [comment]
Correct @jonathanbossenger. Sub-set of data is fine but with no example course or anything, it makes firing up a development environment not easy. We should have a sub-set of data that actually reflects production.
